src/ui/MessageInput/index.tsx

```diff
@@ -394,8 +395,19 @@ const MessageInput = React.forwardRef<HTMLInputElement, MessageInputProps>((prop
       const params = { message: messageText, mentionTemplate };
       onSendMessage(params);
       resetInput(internalRef);
-      // important: keeps the keyboard open -> must add test on refactor
-      textField.focus();
+
+      /**
+       * Note: contentEditable does not work as expected in mobile WebKit (Safari).
+       * @see https://github.com/sendbird/sendbird-uikit-react/pull/1108
+       */
+      if (isMobileIOS(navigator.userAgent)) {
+        if (ghostInputRef.current) ghostInputRef.current.focus();
+        requestAnimationFrame(() => textField.focus());
```

Comment:
> Using setTimeout or similar causes delays due to waiting for the event loop, resulting in the phenomenon of the virtual keyboard going up and down. Therefore, requestAnimationFrame, associated with the browser's cycle of drawing new frames, is used.

I didn't understand this sentence. requestAnimationFrame is using setTimeout with 0 second delay. Did you mean using with 0 second delay won't cause the up and down issue?
```
global.requestAnimationFrame = function (callback) {
  return setTimeout(callback, 0);
};
```

Reply:
No, `requestAnimationFrame` is not using `setTimeout`

requestAnimationFrame is specifically designed to synchronize with the browser’s frame updates.
It queues the callback to execute right before the next repaint, ensuring that updates are smooth and in sync with the browser’s refresh cycle. This makes it much more suitable for animations and dynamic UI updates where timing and visual smoothness are critical.

setTimeout is an API used to delay the execution of a function. It executes the function once after a specified time in milliseconds set by the developer. Since this function runs when the event loop has available time after processing other tasks, the specified delay is the minimum wait time but the actual execution could be delayed further. As a result, it operates independently without synchronizing with other tasks in the browser.

Reply:
That's incredible this thing existed!

Reply:
In this case, what is triggering repaint? `ghostInputRef.current.focus();`?

Reply:
Repaint happens when a browser redraws a web page to show the visual updates resulting from a UI change.